### ISSUE TITLE
Make the "Report a Bug" menu option autofill the issue template with Godot version and system information

### ISSUE THREAD

**Issue by dweremeichik** [godotengine/godot-proposals#3063]
### Describe the project you are working on

Any/Engine

### Describe the problem or limitation you are having in your project

When you click the "Report a Bug" button found under the help menu it takes you to the [issues page](https://github.com/godotengine/godot/issues). From there you have to click "New Issue" then "Get Started" and fill in your information.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Alternatively, the Report a Bug button could take you directly to a new issue and prefill select fields for you such as Godot Version and System Information.

Here is an [example](https://github.com/godotengine/godot/issues/new?template=bug_report.yml&godot_version=3.3.2.stable&system_information=Windows%2010%2C%20OpenGL%20ES%203.0%20Renderer%3A%20GeForce%20GTX%201070%2FPCIe%2FSSE2%2C%20OpenGL%20ES%20Batching%3A%20ON). Notice how the Version and System information fields are already populated.

This feature enhancement would aid in the accuracy of bug reports.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

This line would need to be modified:
https://github.com/godotengine/godot/blob/92c5c46da7e5990ac328437ffea6ec3fcd8a0f10/editor/editor_node.cpp#L2809

http_escape already exists in the engine, and the engine already has access to OS and GPU information.

### If this enhancement will not be used often, can it be worked around with a few lines of script?

Sure, you can enter the information in yourself, but it is error prone. I've submitted bugs with 3.2.3 instead of 3.3.2 because of manual entry.

This would also allow the devs to get more specific information - when someone doesn't fill out system information with a GPU, does it mean they are using integrated graphics, or do the simply not know? This would fix that.

### Is there a reason why this should be core and not an add-on in the asset library?

It's already core; simply an enhancement.

**Comment by Calinou** [comment]
I didn't know the new issue forms could have prefilled fields – this is good to know :slightly_smiling_face: 

Feel free to open a pull request to implement this, if you know how to do it.

**Comment by dweremeichik** [comment]
I would like to start contributing; this is likely a good issue for that. Let me see where I can get with it.

As far as the prefilled fields - I've not seen it advertised officially anywhere, however I had seen it used before with the old system (pre form templates). Under the old system I found [this post] from someones blog and worked backwards from there to get it to work under the new system.

If you take the name of the field and snake case it, or lop off the "issue_form" bit from the textarea id you will get what you need.
![image](https://user-images.githubusercontent.com/4079608/127722869-0831660e-b1cd-444b-8712-7fdb18af651d.png)

Worst case scenario - if github changes this functionality in the future, the url query parameters would just be ignored.

**Comment by LightningAA** [comment]
The only downside I can see with this is might it discourage people from searching for duplicates issues first?

**Comment by Calinou** [comment]
> The only downside I can see with this is might it discourage people from searching for duplicates issues first?

The option to create a new issue directly from the editor is already present, so this particular change wouldn't hurt.

**Comment by Calinou** [comment]
We discussed this proposal in a meeting today, and found several issues with the implementation:

- Opening the issue form directly discourages actively searching for duplicates before opening an issue. We have a *lot* of duplicate issues opened already, so it's important to keep the workload of closing duplicates down. To make things worse, GitHub removed the automatic duplicate finder from its issue creation form a few months ago (we don't know why).
- If we change any of the form titles in the issue template, the IDs will change and the autofill functionality will break.
- If we change the issue template's file name, the issue reporting functionality will break entirely.

Therefore, I've opened https://github.com/godotengine/godot-proposals/issues/4732 which is an alternative take on the original problem.
